sklearn/utils/validation.py

```diff
@@ -9,21 +9,22 @@
 # License: BSD 3 clause
 
 import warnings
-import numbers
 
 import numpy as np
 import scipy.sparse as sp
 from numpy.core.numeric import ComplexWarning
 
 from ..externals import six
 from ..utils.fixes import signature
+from ..utils.fixes import scalar_integer_types
 from .. import get_config as _get_config
 from ..exceptions import NonBLASDotWarning
 from ..exceptions import NotFittedError
 from ..exceptions import DataConversionWarning
 from ..externals.joblib import Memory
 
 
+floating_types = (float, np.floating)
```

Comment:
I think we should rename this to `scalar_floating_types` to make it explicit that this should not be used to validated array `dtype`s.

Also because this tuple of types is not a provided as a backport for old version of numpy but is going to be used to validate scalar paramters in similar ways to `scalar_integer_types`, I think we should both constant here instead of `sklearn.utils.fixes`. Sorry for the back and forth comments.